libs/libc/time/lib_localtime.c

```diff
@@ -676,23 +668,22 @@ static int tzload(FAR const char *name,
       for (i = 0; i < sp->timecnt; ++i)
         {
           int_fast64_t at = stored == 4 ? detzcode(p) : detzcode64(p);
-          sp->types[i] = ((TYPE_SIGNED(time_t)
-                           ? g_min_timet <= at : 0 <= at) &&
-                           at <= g_max_timet);
+          sp->types[i] = at <= g_max_timet;
           if (sp->types[i])
             {
-              if (i && !timecnt && at != g_min_timet)
+              int_fast64_t attime = at < g_min_timet ? g_min_timet : at;
```

Comment:
how int64 value can be less than `INT64_MIN`? I mean what are the values to get this condition evaluated to `true`?

Reply:
@Donny9 probably intended `<=` here?

Reply:
I think that building with `-Wall` and `-Wextra` would give a compiler warning for this?

Reply:
I think even `<=` makes no sense.

Reply:
It would make sense if `g_min_timet` is `INT32_MIN`, but not with `INT64_MIN`. Let me check the max condition as well.

Reply:
> how int64 value can be less than `INT64_MIN`? I mean what are the values to get this condition evaluated to `true`?

Refs to https://github.com/eggert/tz/blob/main/localtime.c#L518. Looks like the check can be removed and using at instead of attime.

Reply:
Yes, but initial code was relying on `time_t` size. The NuttX use `time_t` as `uint32_t`, so MIN and MAX are for 32-bit  unsigned type. But in NuttX implementation we switch to `int_fast64_t` calculations, so I do not see how we can keep referencing to the https://github.com/eggert/tz/blob/main/localtime.c

Reply:
> Yes, but initial code was relying on `time_t` size. The NuttX use `time_t` as `uint32_t`, so MIN and MAX are for 32-bit unsigned type. But in NuttX implementation we switch to `int_fast64_t` calculations, so I do not see how we can keep referencing to the https://github.com/eggert/tz/blob/main/localtime.c

Yes, For Nuttx in `time_t` as `uint32_t`:
```
the TYPE_SIGNED(time_t) is ture,
he g_min_timet is 0, 
the g_max_timet is (time_t)-1.
```
For tzfile "Pacific/Honolulu", ats is `ats = {-2334101314, -1157283000, -1155436200, -880198200, -769395600, -765376200, -712150200,` it's not in the range of `uint32` and `int32` anyway. so need to convert int_fast64_t

Reply:
I'm fine if conversion is needed. What I'm highlighting that some comparisons become not valid if we promote to 64bit

Reply:
> I'm fine if conversion is needed. What I'm highlighting that some comparisons become not valid if we promote to 64bit

The latest patch replaces where only ats needs int64_t, doesn't affect other parts.

Reply:
The code in https://github.com/eggert/tz/blob/main/localtime.c can work correctly only when time_t is 64bits in the internal computation. Since NuttX define time_t as uint32_t, we need modify the internal function/data to int_fast64_t.